packages/console/src/ds-components/TextInput/MultiInput.tsx

```diff
@@ -0,0 +1,171 @@
+  values,
+  onChange,
+  error,
+  placeholder,
+  renderOption = ({ value }) => value,
+}: Props) {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [isInputFocused, setIsInputFocused] = useState(false);
+  const [focusedValueId, setFocusedValueId] = useState<Nullable<string>>(null);
+  const [currentValue, setCurrentValue] = useState('');
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+
+  const isCurrentValueValidDomain = useMemo(() => {
+    return domainRegExp.test(currentValue);
+  }, [currentValue]);
+
+  const handleAdd = (value: string) => {
+    const newValues = [...values, { value, id: generateStandardShortId() }];
+    onChange(newValues);
+    setCurrentValue('');
+    inputRef.current?.focus();
+  };
+
+  const handleDelete = (option: Option) => {
+    onChange(values.filter(({ id }) => id !== option.id));
+  };
+
+  // https://exogen.github.io/blog/focus-state/
+  useEffect(() => {
+    if (document.hasFocus() && inputRef.current?.contains(document.activeElement)) {
+      setIsInputFocused(true);
+    }
+  }, []);
+
+  return (
+    <div
+      className={classNames(
+        styles.input,
+        styles.multiple,
+        isInputFocused && styles.inputFocused,
+        Boolean(error) && styles.error,
+        className
+      )}
+      role="button"
+      tabIndex={0}
+      onKeyDown={onKeyDownHandler(() => {
+        inputRef.current?.focus();
+      })}
+      onClick={() => {
+        inputRef.current?.focus();
+      }}
+    >
+      {values.map((option) => {
+        return (
+          <Tag
+            key={option.id}
+            variant="cell"
+            className={classNames(
+              styles.tag,
+              option.status && styles[option.status],
+              option.id === focusedValueId && styles.focused
+            )}
+            onClick={(event) => {
+              event.stopPropagation();
+            }}
+          >
+            {renderOption(option)}
+            <IconButton
+              className={styles.delete}
+              size="small"
+              onClick={() => {
+                handleDelete(option);
+              }}
+              onKeyDown={onKeyDownHandler(() => {
+                handleDelete(option);
+              })}
+            >
+              <Close className={styles.close} />
+            </IconButton>
+          </Tag>
+        );
+      })}
+      <input
+        ref={inputRef}
+        type="text"
+        placeholder={conditional(values.length === 0 && placeholder && String(t(placeholder)))}
+        value={currentValue}
+        onKeyDown={(event) => {
+          if (event.key === 'Backspace' && currentValue === '') {
+            if (focusedValueId) {
+              onChange(values.filter(({ id }) => id !== focusedValueId));
+              setFocusedValueId(null);
+            } else {
+              setFocusedValueId(values.at(-1)?.id ?? null);
+            }
+            event.stopPropagation();
+          }
```

Comment:
Can be simplified by always removing the last tag from the array when the current value is empty. So the `focusedValueId` is not needed.
Note: might also need to deal with the `delete` key as well as `backspace`

Reply:
See the previous comment, even if we removed `focusedValueId` we still need to record a boolean value to see whether the last tag is focused.